homeassistant/components/tibber/coordinator.py

```diff
@@ -0,0 +1,163 @@
+"""Coordinator for Tibber sensors."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import cast
+
+import tibber
+
+from homeassistant.components.recorder import get_instance
+from homeassistant.components.recorder.models import StatisticData, StatisticMetaData
+from homeassistant.components.recorder.statistics import (
+    async_add_external_statistics,
+    get_last_statistics,
+    statistics_during_period,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfEnergy
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN as TIBBER_DOMAIN
+
+FIVE_YEARS = 5 * 365 * 24
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TibberDataCoordinator(DataUpdateCoordinator[None]):
```

Comment:
Note: I did not migrate `TibberRtDataCoordinator` as that one has a circular dependency with `TibberSensorRT` I wanted to keep this PR small, and I plan to address that one in a separate PR